app/src/components/tables/BlockTable/BlocksTable.tsx

```diff
@@ -29,37 +38,57 @@ const rowCountSelectOptions: SelectOptions[] | null = [
   { value: '20', label: '20 rows' },
 ];
 
-interface BlocksTableProps {
-  readonly total?: number;
-  readonly blocks: ApiData.Block[];
-  readonly showValidators?: boolean;
-  isTableLoading: boolean;
-  onSortingChange?: OnChangeFn<SortingState>;
-  sorting?: SortingState;
-  initialSorting?: SortingState;
-  setIsTableLoading: React.Dispatch<React.SetStateAction<boolean>>;
-}
+const initialSorting: SortingState = [
+  {
+    id: 'height',
+    desc: true,
+  },
+];
 
-export const BlocksTable: React.FC<BlocksTableProps> = ({
-  total,
-  blocks,
-  showValidators,
-  isTableLoading,
-  setIsTableLoading,
-  ...props
-}) => {
+export const BlocksTable: React.FC = () => {
   const { t } = useTranslation();
 
+  const [isTableLoading, setIsTableLoading] = useState(false);
   const [currentTime, setCurrentTime] = useState(Date.now());
   const [showTimestamp, setShowTimestamp] = useState(false);
 
-  const blocksTableOptions = useAppSelector(getBlocksTableOptions);
+  const dispatch = useAppDispatch();
+
+  const blocks = useAppSelector(getBlocks);
   const totalBlocks = useAppSelector(getTotalBlocks);
+  const latestBlock = useAppSelector(getLatestBlock);
+  const blockLoadingStatus = useAppSelector(getBlocksLoadingStatus);
+  const blocksTableOptions = useAppSelector(getBlocksTableOptions);
 
   const totalPages = useMemo(() => {
     return Math.ceil(totalBlocks / blocksTableOptions.pagination.pageSize);
   }, [blocksTableOptions, totalBlocks]);
 
+  const isLoadingPage =
+    blockLoadingStatus !== Loading.Complete && !blocks.length;
+
+  useEffect(() => {
+    dispatch(setInitialStateFromUrlSearchParams());
+  }, [dispatch]);
+
+  useEffect(() => {
+    // updated from WS
+    if (latestBlock) {
+      dispatch(updateBlocksWithLatest({ latestBlock }));
+    }
+  }, [latestBlock, dispatch]);
+
+  useEffect(() => {
+    dispatch(fetchBlocks(blocksTableOptions));
+  }, [dispatch, blocksTableOptions]);
+
+  useEffect(() => {
+    if (isTableLoading || isLoadingPage) {
+      setIsTableLoading(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [blocks]);
+
   useEffect(() => {
```

Comment:
Hmm -- I know this isn't in your PR but I've haven't read this code in depth yet.  Do we have any performance concerns with this being in the top level table component and not creating a separate time component?

I'm just curious if we want to be re-rendering the entire table every second

Reply:
I haven't noticed any performance issues up until this point, but I'm sure there's room for improvement. Do you wanna create a new ticket to investigate this in the future?

Reply:
Yeah -- definitely not in scope for this ticket but we should investigate in another :)